### PR TITLE
src_ext / cold updates

### PR DESCRIPTION
**Pull request by dra27**
- cold compiler to 4.12.0 (allows bootstrapping on Cygwin64)
- src_exts updated to latest versions (apart from dose3 and extlib)
- Tidy-up an old bit of jbuilder needed code in the vendoring system
- Add quite a few `-w -3` so the lib-ext build is quieter
- Update two CI builds to latest point release

### CODE REVIEW

**Comment by dra27** [comment]
How embarrassing. I appear to have revealed the OS I didn't test this on...

File: src_ext/patches/ocamlgraph/restore-4.02-support.patch [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ }
+ 
+diff -Naur a/src/dune b/src/dune
+--- a/src/dune	2020-10-02 13:47:57.000000000 +0000
++++ b/src/dune	2021-03-26 12:33:07.160324274 +0000
+@@ -3,6 +3,7 @@
+ (library
+  (name graph)
+  (public_name ocamlgraph)
++ (flags -w -3)
```
Comment (kit-ty-kate) [inline src_ext/patches/ocamlgraph/restore-4.02-support.patch:20]: shouldn't this be `(flags :standard -w -3)` instead?

Comment (kit-ty-kate) [reply src_ext/patches/ocamlgraph/restore-4.02-support.patch:20]: (same for the rest of the `(flags -w -3)` additions)

Comment (dra27) [reply src_ext/patches/ocamlgraph/restore-4.02-support.patch:20]: Oops, yes!

File: src_ext/Makefile.packages [hunk 2]
```diff
@@ -41,9 +41,13 @@ dose3.pkgbuild: findlib.pkgbuild ocamlbuild.pkgbuild cppo.pkgbuild extlib.pkgbui
...
 mccs.pkgbuild: dune-local.pkgbuild cudf.pkgbuild findlib.pkgbuild
 
+stdlib-shims.pkgbuild: dune-local.pkgbuild
+
+dune-local.pkgbuild: findlib.pkgbuild
+
 opam-file-format-pkg-build:
 	dune build @install -p opam-file-format
-	dune install -p opam-file-format
+	dune install -p opam-file-format opam-file-format
```
Comment (kit-ty-kate) [inline src_ext/Makefile.packages:50]: I'm curious, are these changes really required? Isn't that already implied by the `-p <pkg>` argument?

Comment (dra27) [reply src_ext/Makefile.packages:50]: It turns out they're not (I assumed so, too). You get an error with ocamlgraph complaining that `ocamlgraph_gtk.install` hasn't been built.

**Review by kit-ty-kate — APPROVED** [review]
apart from my last (very minor) comment, if CI is happy this seems all good.

File: src_ext/Makefile.packages (hunk 2 above)
Comment (dra27) [reply src_ext/Makefile.packages:50]: I put that in for the other just for belt-and-braces!

Comment (kit-ty-kate) [reply src_ext/Makefile.packages:50]: I see, interesting
